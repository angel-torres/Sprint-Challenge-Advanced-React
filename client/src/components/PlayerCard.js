import React from 'react';
import { Card } from '@material-ui/core'

class PlayerCard extends React.Component {

    render() {
        return (
            <Card>
                <h1><span>Name:</span> {this.props.player.name}</h1>    
                <p><span>Country:</span> {this.props.player.country}</p>
            </Card>
        )
    }
}

export default PlayerCard;