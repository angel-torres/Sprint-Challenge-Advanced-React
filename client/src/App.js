import React, { useEffect, useState } from 'react';
import './App.css';
import axios from 'axios';
import useLocalStorage from './hooks/useLocalStorage';
import PlayerCard from './components/PlayerCard';


function App() {
  const [players, setPlayers] = useState([])
  const [ setToStorage, removeFromStorage ] = useLocalStorage()

  useEffect(() => {
    axios.get("http://localhost:5000/api/players")
    .then((res) => {
      setPlayers(res.data)
      setToStorage(JSON.stringify(res.data))
    })
    .catch((err) => {
      console.error(err)
    })
  }, [])

  return (
    <div className="App">
      <h1>Players</h1>
      {
        players.map((player) => {
          return (
          <div>
            <PlayerCard player={player}/>
          </div>
          )
        })
      }
    </div>
  );
}

export default App;
