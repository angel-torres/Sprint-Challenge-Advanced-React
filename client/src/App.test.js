import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import App from './App';
import PlayerCard from './components/PlayerCard'
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup)

it('should render \'players\' on screen', () => {
  const { getByText } = render(<App />)

  const element = getByText(/players/i)
  expect(element).toBeInTheDocument()
  expect(element).toBeTruthy()

})

it('render name', () => {
  const { getByText } = render(<PlayerCard player={{name:"John", country:"US"}}/>)

  const element = getByText(/name/i)
  expect(element).toBeInTheDocument()
  expect(element).toBeTruthy()

})

it('matches snapshot', () => {
  const tree = renderer.create(<PlayerCard player={{name:"john", country:"MEX"}}/>).toJSON()

  expect(tree).toMatchSnapshot()
})