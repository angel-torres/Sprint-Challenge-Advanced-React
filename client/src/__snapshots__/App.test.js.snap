// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
>
  <h1>
    <span>
      Name:
    </span>
     
    john
  </h1>
  <p>
    <span>
      Country:
    </span>
     
    MEX
  </p>
</div>
`;
